include/swift/RemoteInspection/ReflectionContext.h

```diff
@@ -1279,6 +1279,10 @@ class ReflectionContext
   std::optional<unsigned>
   computeUnalignedFieldStartOffset(const TypeRef *TR,
                                    remote::TypeInfoProvider *ExternalTypeInfo) {
+    if (auto *objcTR = dyn_cast<ObjCClassTypeRef>(TR))
```

Comment:
Can we move this to after the RemoteMirrors inspection mechanism? Asking for the TypeInfo via the `TypeInfoProvider` can be very expensive in the LLDB case (lots of DWARF parsing).

Reply:
Maybe I'm misunderstanding, but I put this logic at the top because my understanding is that swift remote lookup for `ObjCClassTypeRef` instances will always fail.

Reply:
After tracing the construction of `ObjCClassTypeRef`, these typeref instances  are created only for classes in the `__C` module. Those classes will never have Swift type metadata.

Here is the relevant code:

https://github.com/swiftlang/swift/blob/aa7ff59a136091cbbeff073138563c40abe539bd/include/swift/Demangling/TypeDecoder.h#L622-L626

and

https://github.com/swiftlang/swift/blob/aa7ff59a136091cbbeff073138563c40abe539bd/include/swift/Demangling/TypeDecoder.h#L432-L436